### ISSUE TITLE
Error in vignettes on PPC: figure margins too large

### ISSUE THREAD

**Issue by barracuda156** [RobinHankin/elliptic#6]
```
* checking tests ...
  Running ‘aaa.R’
 OK
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in ‘inst/doc’ ... OK
* checking running R code from vignettes ...
  ‘ellipticpaper.Rnw’... failed
  ‘residuetheorem.Rnw’... OK
 ERROR
Errors in running code in vignettes:
when running code in ‘ellipticpaper.Rnw’
  ...

> png("wp_figure.png", width = 800, height = 800)

> persp(x, y, limit(Re(f)), xlab = "Re(z)", ylab = "Im(z)", 
+     zlab = "Re(P(z))", theta = 30, phi = 30, r = 1e+09, border = NA, 
+     shade = 0.8 .... [TRUNCATED] 

  When sourcing ‘ellipticpaper.R’:
Error: figure margins too large
Execution halted
```

```
Attaching package: ‘elliptic’

The following objects are masked from ‘package:stats’:

    sd, sigma

The following object is masked from ‘package:base’:

    is.primitive

Loading required package: emulator
Loading required package: mvtnorm
Loading required package: calibrator

Error: processing vignette 'ellipticpaper.Rnw' failed with diagnostics:
 chunk 21 (label = wp_figure_plot) 
Error in plot.new() : figure margins too large

--- failed re-building ‘ellipticpaper.Rnw’
```

macOS 10.6.8 Rosetta, R 4.2.2, gcc 12.2.0

**Comment by RobinHankin** [comment]
I can't reproduce the error, it is from an old version of the package as `ellipticpaper.Rnw` does not exist any more.  Also, let me know how you triggered it, was it `R CMD check` on the command line, or some other way?

**Comment by barracuda156** [comment]
@RobinHankin Thank you for responding. I took the package from CRAN, since there are no releases here. CRAN does seem to have quite an old version of it: https://cran.r-project.org/web/packages/elliptic/index.html (but I actually assumed there were no updates since then; I can fetch from git repo instead, of course).

Yes, test are run with `R CMD check`: https://github.com/macports/macports-ports/blob/654c7fa2c51b459eb0738206d291668ffb53f3c0/_resources/port1.0/group/R-1.0.tcl#L143-L145

TBH, I got this error several times: “Error: figure margins too large”. So it is definitely not specific to the code here. It likely results from 32-bits limitations (?).
However, I do not have a solution to it. So if there are any ideas, it would be very helpful.
